src/LightGBM.jl

```diff
@@ -39,7 +40,16 @@ end
 
 
 function __init__()
-    LGBM_library[] = Libdl.dlopen(find_library("lib_lightgbm", [@__DIR__]))
+    if Sys.isapple()
+        # for macOS users alternative LightGBM_jll - binaries compiled with Binary Builder
+        # particularly to address the issues raised for M1/aarch64-apple-darwin
+        @eval using LightGBM_jll
+        LGBM_library[] = Libdl.dlopen(find_library("lib_lightgbm", [@__DIR__]))
```

Comment:
@giordano is it correct that
```julia
LGBM_library[] = Libdl.dlopen(find_library("lib_lightgbm", [@__DIR__]))
```
all that is necessary to get the `LightGBM_jll` binaries?

Reply:
The library is automatically dlopened when `using` the package, I don't understand the point of this complication.  If you need only the path, it's `LightGBM_jll.lib_lightgbm_path`, otherwise I don't know what's happening here.

Reply:
> The library is automatically dlopened when `using` the package, I don't understand the point of this complication. If you need only the path, it's `LightGBM_jll.lib_lightgbm_path` (see https://docs.binarybuilder.org/stable/jll/#LibraryProduct), otherwise I don't know what's happening here.

Thank you @rikhuijzer and @giordano for your comments. Correct me if I'm wrong but by only using LightGBM_jll it is no longer possible to provide custom built binaries because LightGBM_jll will provide the prebuilt "lib_lightgbm" binaries for the given architecture without giving the user a way to alter it, if needed.  Currently, it is possible to still built a lightGBM binary from source and for e.g. get the CUDA version of lightgbm given that the original gpu build is OpenCL (and I think this is from lightgbm v4 not v3 anyway so for v3 either way needs to be built from scratch).

Reply:
https://docs.binarybuilder.org/stable/jll/#Overriding-the-artifacts-in-JLL-packages